explorer/interpreter/value.h

```diff
@@ -935,6 +935,25 @@ struct ImplConstraint {
   Nonnull<const InterfaceType*> interface;
 };
 
+// A constraint that requires an intrinsic property of a type.
+struct IntrinsicConstraint {
+  // The type that is required to satisfy the intrinsic property.
+  Nonnull<const Value*> type;
+  // The kind of the intrinsic property.
+  enum Kind {
+    // `type` intrinsically implicitly converts to `parameters[0]`.
+    // TODO: Split this up into finer-grained checks.
+    ImplicitAs,
```

Comment:
What other kinds are you expecting? Is this a TODO to refactor the kind? It just seems a little weird to me right now as a single-value enum.

Reply:
I'm expecting lots of other kinds; these intrinsic constraints serve the same purpose as built-in type traits in C++, of which there are [many](https://en.cppreference.com/w/cpp/header/type_traits).

The TODO is to split the `ImplicitAs` intrinsic constraint into more specific constraints (eg, one for derived-to-base pointer conversions, one for struct conversions, ...), which @chandlerc requested as our eventual design here.

Reply:
Suggested rephrasing of the comment to be more explicit.